ts/sagas/wallet.ts

```diff
@@ -308,28 +305,35 @@ function* startOrResumeAddCreditCardSaga(
     // Even though this step is optional, in practice Pagopa will always
     // require the 3DS checkout for cards that gets added to the wallet.
     //
-
     const urlCheckout3ds =
       state.creditCardVerification.value.data.urlCheckout3ds;
-    const pagoPaToken = pmSessionManager.get();
-
-    if (pot.isNone(state.creditCardCheckout3ds)) {
-      if (urlCheckout3ds !== undefined && pagoPaToken.isSome()) {
-        yield put(
-          creditCardCheckout3dsRequest({
-            urlCheckout3ds,
-            paymentManagerToken: pagoPaToken.value
-          })
-        );
-        yield take(getType(creditCardCheckout3dsSuccess));
-        // all is ok, continue to the next step
-        continue;
-      } else {
-        // if there is no need for a 3ds checkout, simulate a success checkout
-        // to proceed to the next step
-        yield put(creditCardCheckout3dsSuccess("done"));
-        continue;
+    try {
+      // Request a new token to the PM. This prevent expired token during the webview navigation.
+      // If the request for the new token fails a new Error is catched, the step fails and we exit the flow.
+      const pagoPaToken = yield call(pmSessionManager.getNewToken());
+      if (pot.isNone(state.creditCardCheckout3ds)) {
+        if (urlCheckout3ds !== undefined && pagoPaToken.isSome()) {
```

Comment:
The else case is about
```
// if there is no need for a 3ds checkout, simulate a success checkout
// to proceed to the next step
```
this is actually not true because the `if` condition changed with `pagoPaToken.isSome()` & condition

we should handle and track when is not possibile to renew the wallet token
```typescript
[...]

if (urlCheckout3ds !== undefined) {
    if (pagoPaToken.isSome()) {
        yield put(
            creditCardCheckout3dsRequest({
                urlCheckout3ds,
                paymentManagerToken: pagoPaToken.value
            })
        );
        yield take(getType(creditCardCheckout3dsSuccess));
        // all is ok, continue to the next step
        continue;
    } else {
        if (action.payload.onFailure) {
            yield call(
                mixpanelTrack,
                getType(addWalletNewCreditCardFailure), {
                    reason: "cannot refresh wallet token"
                }
            );
            action.payload.onFailure();
        }
        return;
    }
} else {
    // if there is no need for a 3ds checkout, simulate a success checkout
    // to proceed to the next step
    yield put(creditCardCheckout3dsSuccess("done"));
    continue;
}

[...]
```

Reply:
Fixed in [94c54a0](https://github.com/pagopa/io-app/pull/2490/commits/94c54a0a3b0c2982d59eceea7da4e38a232059da)